addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -172,20 +228,123 @@ function ComposerFactory({ Entity }) {
             }
         }
 
-        /**
-         * @param {Object} param0
-         * @param {string} param0.textInputContent
-         * @param {integer} param0.textInputCursorEnd
-         * @param {integer} param0.textInputCursorStart
-         */
-        saveTextInput({ textInputContent, textInputCursorEnd, textInputCursorStart }) {
-            this.update({ textInputContent, textInputCursorEnd, textInputCursorStart });
+        setFirstSuggestedPartnerActive() {
+            if (this._allSuggestedPartners[0]) {
+                this.update({ activeSuggestedPartner: [['link', this._allSuggestedPartners[0]]] });
+            }
+        }
+
+        setLastSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                this.update({
+                    activeSuggestedPartner: [['link', this._allSuggestedPartners[this._allSuggestedPartners.length - 1]]]
+                });
+            }
+        }
+
+        setNextSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                const activeSuggestedPartnerIndex = this._allSuggestedPartners.findIndex(
+                    suggestedPartner => suggestedPartner === this.activeSuggestedPartner
+                );
+                if (
+                    activeSuggestedPartnerIndex !== this._allSuggestedPartners.length - 1) {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[activeSuggestedPartnerIndex + 1]]]
+                    });
+                } else {
+                    this.update({ activeSuggestedPartner: [['link', this._allSuggestedPartners[0]]] });
+                }
+            }
+        }
+
+        setPreviousSuggestedPartnerActive() {
+            if (this._allSuggestedPartners.length > 0) {
+                const activeSuggestedPartnerIndex = this._allSuggestedPartners.findIndex(
+                    suggestedPartner => suggestedPartner === this.activeSuggestedPartner
+                );
+                if (activeSuggestedPartnerIndex !== 0) {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[activeSuggestedPartnerIndex - 1]]]
+                    });
+                } else {
+                    this.update({
+                        activeSuggestedPartner: [['link', this._allSuggestedPartners[this._allSuggestedPartners.length - 1]]]
+                    });
+                }
+            }
         }
 
         //----------------------------------------------------------------------
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         *
+         * @private
+         */
+        _computeAllSuggestedPartners() {
+            return [['replace', this.mainSuggestedPartners.concat(this.extraSuggestedPartners)]];
```

Comment:
I would add a comment here, just to explicit it at least somewhere in the code, that this works only if we can assume that one partner cannot be on both lists at the same time.